dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/UninitializedEventDescriptor.scala

```diff
@@ -0,0 +1,65 @@
+package org.bitcoins.dlc.oracle
+
+import org.bitcoins.core.number._
+import org.bitcoins.core.protocol.tlv._
+import org.bitcoins.crypto.SchnorrNonce
+
+sealed trait UninitializedEventDescriptor[+T <: EventDescriptorTLV] {
+  def noncesNeeded: Int
+
+  def addNonce(nonce: SchnorrNonce): T = addNonces(Vector(nonce))
+  def addNonces(nonces: Vector[SchnorrNonce]): T
+}
+
+case class UninitializedExternalEventDescriptorV0(externalName: String)
+    extends UninitializedEventDescriptor[ExternalEventDescriptorV0TLV] {
+
+  override def addNonces(
+      nonces: Vector[SchnorrNonce]): ExternalEventDescriptorV0TLV = {
+    require(nonces.size == 1, "ExternalEventDescriptorV0 can only have 1 nonce")
+    ExternalEventDescriptorV0TLV(nonces.head, externalName)
+  }
+
+  override val noncesNeeded: Int = 1
```

Comment:
Why is this hard coded?

Reply:
I think it should be left abstract so we don't forget to override this when we add more case classes to our ADT in the future.

Reply:
Currently in the spec external event descriptors only have 1 nonce

Reply:
Oh i see why you are doing it now, but this isn't great. Hmm...

Reply:
https://github.com/discreetlogcontracts/dlcspecs/pull/55#discussion_r504176719

Reply:
Looks like this was removed actually